### ISSUE TITLE
Fresh Plastid install is incompatible with pysam version

### ISSUE THREAD

**Issue by lparsons** [bioconda/bioconda-recipes#15774]
It seems that the version of pysam isn't pinned when installing plastid. Since pysam isn't explicitly on this list: https://github.com/bioconda/bioconda-utils/blob/master/bioconda_utils/bioconda_utils-conda_build_config.yaml I'm guessing it won't get pinned automatically. I can think of two ways to handle this:
1. Add pysam to the list of automatically pinned packages
2. Add htslib directly to the dependency list of plastid (and really any other package that lists pysam).

Or perhaps I'm totally off. @bioconda/core any thoughts?

**Comment by lparsons** [comment]
Running tests on a PR now. Was just checking to see if a new version was released (no :disappointed: ). There is a fix for numpy 1.13 compatibility committed, but not in a "release" yet.

**Comment by lparsons** [comment]
Hmmm... It's also blacklisted "takes hours to build": https://github.com/bioconda/bioconda-recipes/blob/master/build-fail-blacklist#L321

I don't think that's true (at least not for me, so I'll try removing that).

**Comment by dpryan79** [comment]
That wording was likely from me and may have been a transient problem (I was trying to get things rebuilt in the `bulk` branch and a few recipes would just sit there doing nothing for 4 hours).

**Comment by lparsons** [comment]
Weird, seems to have worked, but a simple test of command line script (`counts_in_region -h`) seems to be hanging. I can't test on my local machine (just says "nothing to do"). I'm willing to bet it's working, but not sure how to test it. I know it *used* to work...

**Comment by epruesse** [comment]
Are you able to click the "rerun with ssh" button on circleci? You have to log in. It might be limited to core members.

**Comment by lparsons** [comment]
No, it says "You need write permissions to trigger builds". If someone can give me an idea of what happens when these commands are called, I can try to fix it. Or perhaps give me an idea of how to trigger a build locally that I can test with? Not sure how to do that either.

**Comment by lparsons** [comment]
Hmmm... Seems CircleCi's website is a bit wonky. When I view the cancelled build I got that message, but was able to click the "rerun with ssh" button when I went to the "running" build. I'll see if I can figure that out.

**Comment by lparsons** [comment]
OK, I reran with ssh enabled, but I'm getting a permission denied (publickey) error. I believe things are setup OK on GitHub (at least `ssh -T git@github.com` seems to work).

**Comment by epruesse** [comment]
In theory, it should just work with the key you use for committing.

**Comment by lparsons** [comment]
@epruesse I am now able to login to CircleCi using SSH (no idea what changed, perhaps time, it was a new key I was using). However, I have no idea where to find anything there. I can see the repo, but none of the build artifacts. Perhaps in some docker container somewhere?

**Comment by epruesse** [comment]
Ah dang, yes. I keep forgetting. That means that it doesn't help much to check on CircleCI - you can just run in local docker. Sorry.

You can try to "catch" the docker while it's running. Just SSH in, wait until it gets stuck, then use `docker ps` to find it and `docker exec ...` to run a bash in there.

**Comment by epruesse** [comment]
What was the problem with running `bioconda-utils build` locally again? It should with `--docker` and/or `--mulled` give you just the same results as here.

**Comment by lparsons** [comment]
I was able to check things out in the docker container, which led me to the issue (fixed by a patch in #15775). Building locally always gives me the message `Nothing to be done` even with the build number incremented, etc.
```
Docker image digest: sha256:7b3f88035e74739f72373aff135919f3a42cf59006b2832ec6293afdf727c5c1
====>> Spin up Environment
Build-agent version 0.1.1250-22bf9f5d (2018-12-12T11:32:15+0000)
Starting container bioconda/bioconda-utils-build-env
  using image bioconda/bioconda-utils-build-env@sha256:a7b274f9daac0928341ff3c2db0b357e24d97b14777dd810aaee8e053a8f7eb3

Using build environment variables:
  BASH_ENV=/tmp/.bash_env-localbuild-1560356322
  CI=true
  CIRCLECI=true
  CIRCLE_BRANCH=plastid_pysam_pinning
  CIRCLE_BUILD_NUM=
  CIRCLE_JOB=build
  CIRCLE_NODE_INDEX=0
  CIRCLE_NODE_TOTAL=1
  CIRCLE_REPOSITORY_URL=git@github.com:lparsons/bioconda-recipes.git
  CIRCLE_SHA1=119f6b9b15afa9941be5468d60e23764528ba54e
  CIRCLE_SHELL_ENV=/tmp/.bash_env-localbuild-1560356322
  CIRCLE_WORKING_DIRECTORY=~/project

====>> Checkout code
  #!/bin/bash -eo pipefail
mkdir -p /root/project && cp -r /tmp/_circleci_local_build_repo/. /root/project
====>> Setup ssh
  #!/bin/bash -eo pipefail
mkdir -p ~/.ssh
ssh-keyscan -H github.com >> ~/.ssh/known_hosts

# github.com SSH-2.0-babeld-f3847d63
====>> echo ". /opt/conda/etc/profile.d/conda.sh" >> $BASH_ENV
  #!/bin/bash -eo pipefail
echo ". /opt/conda/etc/profile.d/conda.sh" >> $BASH_ENV
====>> echo "conda activate" >> $BASH_ENV
  #!/bin/bash -eo pipefail
echo "conda activate" >> $BASH_ENV
====>> Download common definitions
  #!/bin/bash -eo pipefail
curl -s https://raw.githubusercontent.com/bioconda/bioconda-common/master/common.sh > .circleci/common.sh

====>> Setup bioconda-utils
  #!/bin/bash -eo pipefail
.circleci/setup.sh
remote: Enumerating objects: 107, done.        
remote: Counting objects: 100% (107/107), done.        
remote: Compressing objects: 100% (9/9), done.        
remote: Total 169 (delta 98), reused 105 (delta 98), pack-reused 62        
Receiving objects: 100% (169/169), 627.45 KiB | 9.09 MiB/s, done.
Resolving deltas: 100% (100/100), completed with 37 local objects.
From https://github.com/bioconda/bioconda-recipes
 * [new branch]          master     -> __upstream__BbEeXV3BLA/master
bioconda-utils is /opt/conda/bin/bioconda-utils
Warning: Permanently added the RSA host key for IP address '192.30.253.112' to the list of known hosts.
Permission denied (publickey).
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
====>> Linting
  #!/bin/bash -eo pipefail
bioconda-utils lint recipes config.yml \
--loglevel debug --full-report \
--git-range master HEAD

16:18:57 BIOCONDA DEBUG get_recipes(recipes, package='['*']'): *
16:18:57 BIOCONDA INFO Considering total of 6755 recipes.
16:18:57 GIT DEBUG Popen(['git', 'remote', 'get-url', '--all', 'origin'], cwd=/root/project, universal_newlines=False, shell=None)
16:18:57 GIT DEBUG Popen(['git', 'remote', 'get-url', '--all', 'upstream'], cwd=/root/project, universal_newlines=False, shell=None)
16:18:57 GIT DEBUG Popen(['git', 'remote', 'get-url', '--all', 'hepcat72'], cwd=/root/project, universal_newlines=False, shell=None)
16:18:57 ASYNCIO DEBUG Using selector: EpollSelector
16:18:57 GIT DEBUG Popen(['git', 'merge-base', 'master', 'HEAD'], cwd=/root/project, universal_newlines=False, shell=None)
16:18:57 GIT DEBUG Popen(['git', 'cat-file', '--batch-check'], cwd=/root/project, universal_newlines=False, shell=None)
16:18:58 GIT DEBUG Popen(['git', 'diff-tree', '21f4ac4384d9bcddf4e25a0426f3e5ade92b6328', 'HEAD', '-r', '--abbrev=40', '--full-index', '-M', '--raw', '--no-color'], cwd=/root/project, universal_newlines=False, shell=None)
16:18:58 GIT DEBUG Popen(['git', 'merge-base', 'master', 'HEAD'], cwd=/root/project, universal_newlines=False, shell=None)
16:18:58 GIT DEBUG Popen(['git', 'cat-file', '--batch'], cwd=/root/project, universal_newlines=False, shell=None)
16:18:58 BIOCONDA INFO Constraining to 1 git modified recipes (plastid).
All checks OK
====>> Testing
  #!/bin/bash -eo pipefail
bioconda-utils build recipes config.yml \
--git-range master HEAD

16:19:07 BIOCONDA INFO Recipes modified according to git: recipes/plastid
16:19:07 BIOCONDA INFO Nothing to be done.
Success!
```
